runtime/jvmti/jvmtiHook.c

```diff
@@ -811,6 +859,16 @@ hookRequiredEvents(J9JVMTIEnv * j9env)
 		return 1;
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+	if (hookRegister(vmHook, J9HOOK_VM_VIRTUAL_THREAD_STARTED, jvmtiHookVirtualThreadCreated, OMR_GET_CALLSITE(), j9env)) {
```

Comment:
The new hook needs to be triggered from `VirtualThread.notifyJvmtiMountBegin(firstMount == true)`: https://github.com/eclipse-openj9/openj9/pull/15517.

Reply:
We can introduce `VirtualThread.notifyJvmtiMountBegin` with the hook trigger in this PR. Later, merge conflicts can be resolved in https://github.com/eclipse-openj9/openj9/pull/15517.

Reply:
Not needed as thread data will be lazy allocated